pkgs/development/python-modules/pytorch-ocl/default.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  cmake,
+  torch,
+  ocl-icd,
+  opencl-clhpp,
+  sqlite,
+  python,
+}:
+buildPythonPackage rec {
+  pname = "pytorch-ocl";
+  version = "0.2.0";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "artyom-beilis";
+    repo = "pytorch_dlprim";
+    rev = "${version}";
+    hash = "sha256-N0NjpRWFAYeh2GQu8PPTtf4ZG2Jp9fgttjCbWOYQOmg=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  dependencies = [
+    torch
+    ocl-icd
+    opencl-clhpp
+    sqlite
+  ];
+
+  dontUseSetuptoolsBuild = true;
+  dontUsePipInstall = true;
+  dontUseSetuptoolsCheck = true;
+
+  postInstall = ''
+    mkdir -p $out/${python.sitePackages}
+    mv $out/python/* $out/${python.sitePackages}
+    rmdir $out/python
+  '';
+
+  pythonImportsCheck = [
+    "pytorch_ocl"
+  ];
+
+  meta = with lib; {
```

Comment:
Considering that, but could you give me more information about why going without the `with` statement is better? I'm asking because I had PRs where the exact opposite (moving `lib` outside into a `with` statement) was requested in reviews.

Reply:
https://nix.dev/guides/best-practices.html

I just read it here where they say to avoid using 'with', smaller scopes are supposedly fine, so I guess you can ignore it if you don't feel like changing this part

Reply:
Given the large number of packages using `meta = with lib; {...` currently, I guess it's indeed fine, and will stick with it for now. Anyway thanks for the pointer.